test/test_create_observations.js

```diff
@@ -1,84 +1,300 @@
+    Datastream: { iotId: 1 },
+    components: [ phenomenonTime, result, featureOfInterestId ],
+    'dataArray@iot.count': 2,
+    dataArray: [ [ '2010-12-23T10:20:00:000Z', 1 ],
+                 [ '2010-12-23T10:21:00:000Z', 1 ] ]
+  }];
+
+  const CreateObservationsRequestIotCountDataArrayLengthMismatch = [{
+    Datastream: { iotId: 1 },
+    components: [ phenomenonTime, result ],
+    'dataArray@iot.count': 1,
+    dataArray: [ [ '2010-12-23T10:20:00:000Z', 1 ],
+                 [ '2010-12-23T10:21:00:000Z', 1 ] ]
+  }];
+
+  const CreateObservationsRequestInvalidDatastream = [{
+    Datastream: { iotId: 1 },
+    components: [ phenomenonTime, result, featureOfInterestId ],
+    'dataArray@iot.count': 2,
+    dataArray: [ [ '2010-12-23T10:20:00.000Z', 20, 1 ],
+                 [ '2010-12-23T10:21:00.000Z', 21, 1 ] ]
+  }, {
+    Datastream: { iotId: 99 },
+    components: [ phenomenonTime, result, featureOfInterestId ],
+    'dataArray@iot.count': 2,
+    dataArray: [ [ '2010-12-23T10:20:00.000Z', 20, 1 ],
+                 [ '2010-12-23T10:21:00.000Z', 21, 1 ] ]
+  }];
+
+  const CreateObservationsRequestNotArray = {
+    Datastream: { iotId: 1 },
+    components: [ phenomenonTime, result, featureOfInterestId ],
+    'dataArray@iot.count': 2
+  };
 
-  it('Components, dataArray mismatch', done => {
-    postError(done, CreateObservationsRequestComponentsDataArrayMismatch, 400,
-              '\'components\' don\'t match \'dataArray\' values.');
+  [CreateObservationsRequest[0],
+   CreateObservationsRequestMissingDatastream[0],
+   CreateObservationsRequestMissingResult[0],
+   CreateObservationsRequestMissingPhenomenonTime[0],
+   CreateObservationsRequestInvalidFeatureOfInterestComponent[0],
+   CreateObservationsRequestUnknownComponent[0],
+   CreateObservationsRequestNotArray,
+   CreateObservationsRequestNoFoI[0] ].forEach(body => {
+    let valueArray = [];
+    for (let i = 0; i < body[dataArrayIotCount]; i++) {
+      // Create different `phenomenonTime`s and `result`s for each request.
+      // `FeatureOfInterest/id` here is a placeholder that will be
+      // set by the test when the FeatureOfInterest instance is created.
+      valueArray.push(['2010-12-23T10:2' + i + ':00.000Z', 20 + i, 1]);
+    }
+    body.dataArray = [];
+    body.dataArray = body.dataArray.concat(valueArray);
   });
 
-  it('Input is not an array', done => {
-    postError(done, CreateObservationsRequestNotArray, 400,
-              'Input must be an array');
+  describe('CreateObservations tests', () => {
+
+    beforeEach(done => {
+      models.sequelize.transaction(transaction => {
+        return Promise.all(Object.keys(entities).map(name => {
+          return models[name].destroy({ transaction, where: {} });
+        }));
+      }).then(() => done());
+    });
+
+    it('Create observations specifying FeatureOfInterest', done => {
+      let body = JSON.parse(JSON.stringify(CreateObservationsRequest));
```

Comment:
You are right about the array of objects comment. Ignore the `[0]` part, please.